ruma-api-macros/src/api/request.rs

```diff
@@ -49,9 +49,27 @@ impl Request {
 
             let field_name = &field.ident;
 
-            quote! {
-                #import_path::exports::http::header::#header_name,
-                #import_path::exports::http::header::HeaderValue::from_str(self.#field_name.as_ref())?
+            match &field.ty {
+                syn::Type::Path(syn::TypePath { path: syn::Path { segments, .. }, .. })
+                    if segments.iter().any(|seg| seg.ident == "Option") =>
+                {
+                    quote! {
+                        let req_builder = if let Some(header_val) = self.#field_name.as_ref() {
+                            req_builder.header(
+                                #import_path::exports::http::header::#header_name,
+                                #import_path::exports::http::header::HeaderValue::from_str(header_val)?,
+                            )
+                        } else {
+                            req_builder
+                        };
```

Comment:
This can just be written as

```suggestion
                        if let Some(header_val) = self.#field_name.as_ref() {
                            req_builder = req_builder.header(
                                #import_path::exports::http::header::#header_name,
                                #import_path::exports::http::header::HeaderValue::from_str(header_val)?,
                            );
                        }
```

right?